From: "Yan, Zheng" <zyan@redhat.com>
Date: Thu, 28 Apr 2016 22:56:44 +0800
Subject: ceph: don't forbid marking directory complete after forward seek
Git-commit: 076c40f18d10489e29c515bf5936952830df5e16
Patch-mainline: v4.7-rc1
References: FATE#322288

Forward seek within same frag does not update fi->last_name, it will
not affect contents of later readdir reply. So there is no need to
forbid marking directory complete

Signed-off-by: Yan, Zheng <zyan@redhat.com>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 fs/ceph/dir.c |    5 -----
 1 file changed, 5 deletions(-)

--- a/fs/ceph/dir.c
+++ b/fs/ceph/dir.c
@@ -547,7 +547,6 @@ static loff_t ceph_dir_llseek(struct fil
 {
 	struct ceph_file_info *fi = file->private_data;
 	struct inode *inode = file->f_mapping->host;
-	loff_t old_offset = ceph_make_fpos(fi->frag, fi->next_offset);
 	loff_t retval;
 
 	mutex_lock(&inode->i_mutex);
@@ -574,10 +573,6 @@ static loff_t ceph_dir_llseek(struct fil
 		if (need_reset_readdir(fi, offset)) {
 			dout("dir_llseek dropping %p content\n", file);
 			reset_readdir(fi);
-		} else if (fpos_cmp(offset, old_offset) > 0) {
-			/* reset dir_release_count if we did a forward seek */
-			fi->dir_release_count = 0;
-			fi->readdir_cache_idx = -1;
 		}
 	}
 out:
